### PR TITLE
docs: add a mojave dark mode guide

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This is very much a loose first pass, feedback welcome 👍 

Notes: no-notes

### CODE REVIEW

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Mojave Dark Mode
+
+In macOS 10.14 Mojave Apple introduced a new [system wide dark mode](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/dark-mode/)
+for all macOS computers.  By default Electron apps do not automatically adjust their UI and native interfaces
+to the dark mode setting when it's enabled, this is primarily due to Apple's own guidelines saying you **shouldn't**
+use the dark mode native interfaces if the your app's own interfaces don't support dark mode themsleves.
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: * if the your
* themsleves

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Mojave Dark Mode
+
+In macOS 10.14 Mojave Apple introduced a new [system wide dark mode](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/dark-mode/)
+for all macOS computers.  By default Electron apps do not automatically adjust their UI and native interfaces
+to the dark mode setting when it's enabled, this is primarily due to Apple's own guidelines saying you **shouldn't**
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: run-on sentence. Better to start a new sentence: "This is due to..."

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+use the dark mode native interfaces if the your app's own interfaces don't support dark mode themsleves.
+
+If your app does have a dark mode though, you can make your Electron app follow the system wide dark mode setting.
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: * s/though//
* s/system wide/system-wide/

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+If you are using [`electron-packager`](https://github.com/electron-userland/electron-packager) or
+[`electron-forge`](https://github.com/electron-userland/electron-forge) you can set the
+[`darwinDarkModeSupport`](https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#darwindarkmodesupport)
+option when packaging and this key will be set for you.
+
+## Automatically updating your own interfaces
+
+If your app has it's own dark mode you should toggle it on and off in sync with the system's dark mode setting.  You can do
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: * s/it's/its/

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+this by listening for the theme changed event on Electorn's `systemPreferences` module.  E.g.
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: * s/Electorn/Electron/

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Mojave Dark Mode
+
+In macOS 10.14 Mojave Apple introduced a new [system wide dark mode](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/dark-mode/)
```
Comment (ckerr) [inline docs/tutorial/mojave-dark-mode-guide.md]: s/system wide/system-wide/

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  ...
+  <key>NSRequiresAquaSystemAppearance</key>
+  <false />
+  ...
+</dict>
+</plist>
+```
+
+If you are using [`electron-packager`](https://github.com/electron-userland/electron-packager) or
+[`electron-forge`](https://github.com/electron-userland/electron-forge) you can set the
```
Comment (malept) [inline docs/tutorial/mojave-dark-mode-guide.md]: Maybe indicate the versions of Electron Packager (>= 12.2.0) / Electron Forge (>= 6) that have support for this flag?

**Comment by MarshallOfSound** [comment]
@ckerr @malept This is ready for another look

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
+# Mojave Dark Mode
+
+In macOS 10.14 Mojave Apple introduced a new [system-wide dark mode](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/dark-mode/)
```
Comment (malept) [inline docs/tutorial/mojave-dark-mode-guide.md]:
```suggestion
In macOS 10.14 Mojave, Apple introduced a new [system-wide dark mode](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/dark-mode/)
```

File: docs/tutorial/mojave-dark-mode-guide.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+for all macOS computers.  By default Electron apps do not automatically adjust their UI and native interfaces
+to the dark mode setting when it's enabled. This is primarily due to Apple's own guidelines saying you **shouldn't**
+use the dark mode native interfaces if your app's own interfaces don't support dark mode themselves.
+
+If your app does have a dark mode, you can make your Electron app follow the system-wide dark mode setting.
+
+## Automatically updating the native interfaces
+
+"Native Interfaces" include the file picker, window border, dialogs, context menus and more; basically anything where
+the UI comes from macOS and not your app.  In order to make these interfaces update to dark mode automatically you need
```
Comment (malept) [inline docs/tutorial/mojave-dark-mode-guide.md]:
```suggestion
the UI comes from macOS and not your app.  In order to make these interfaces update to dark mode automatically, you need
```

**Pull request merged**